src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/AnotherUserProfileActivity.java

```diff
@@ -0,0 +1,201 @@
+     */
+    private TextView followThisAccountLabel;
+
+    /**
+     * Text view for the All habits label
+     */
+    private TextView allHabitsLabel;
+
+    /**
+     * List View for public habits
+     */
+    private ListView habitList;
+
+    /**
+     * Class which handles viewing and following another user
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_another_user_profile);
+
+        // Set up the back button
+        ImageButton back = findViewById(R.id.another_user_profile_back_button);
+        back.setOnClickListener(v -> {
+            finish();
+        });
+
+        // Retrieve the logged-in user's userid
+        String currentUserId = AuthorizationService.getInstance().getCurrentUserId();
+
+        // Get the another user's ID from the previous activity
+        Intent intent = getIntent();
+        String anotherUserID = intent.getStringExtra("USER_ID");
+
+        // Retrieve the specific views
+        name = (TextView) findViewById(R.id.name);
+        username = (TextView) findViewById(R.id.username);
+        followBtn = (Button) findViewById(R.id.follow_button);
+        followingLabel = (TextView) findViewById(R.id.following_label);
+        followThisAccountLabel = (TextView) findViewById(R.id.follow_this_account_label);
+        allHabitsLabel = (TextView) findViewById(R.id.all_habits_label);
+        habitList = (ListView) findViewById(R.id.current_user_habit_listview);
+
+
+        // Set up the controller
+        userController = new UserController();
+
+        // Insert the name and the username
+        userController.readUser(anotherUserID, retrievedAnotherUser -> {
+            name.setText(retrievedAnotherUser.getFirstName() + " " + retrievedAnotherUser.getLastName());
+            username.setText("@" + retrievedAnotherUser.getUsername());
+        });
+
+        // Display (Not following and hasn't requested, Requested, or Following) scenarios based on Social Map values of current user and user on screen
+        userController.readUser(currentUserId, retrievedCurrentUser -> {
+            userController.readUser(anotherUserID, retrievedAnotherUser -> {
+                // Get the Social Map values accordingly and set it to null if it doesn't exist
+                Integer valueOfAnotherUserInCurrentUserSocialMap = retrievedCurrentUser.getSocial().get(anotherUserID) == null ? null : ((Number) retrievedCurrentUser.getSocial().get(anotherUserID)).intValue();
+                Integer valueOfCurrentUserInAnotherUserSocialMap = retrievedAnotherUser.getSocial().get(currentUserId) == null ? null : ((Number) retrievedAnotherUser.getSocial().get(currentUserId)).intValue();
+                
+                if ((isNull(valueOfAnotherUserInCurrentUserSocialMap) || valueOfAnotherUserInCurrentUserSocialMap == 0) && isNull(valueOfCurrentUserInAnotherUserSocialMap)) {
```

Comment:
Would be good to add a comment which explains the meanings of 0,1,2 for people who are not familiar with this design.